packages/starknet-snap/src/ui/utils.tsx

```diff
@@ -80,3 +92,149 @@ export async function generateExecuteTxnFlow(
     },
   });
 }
+
+/**
+ * Update the interface for a ExecuteTxnUI
+ *
+ * @param id - Interface Id
+ * @param request - TransactionRequest
+ * @param errors
+ * @param errors.errors
+ */
+export async function updateExecuteTxnFlow(
+  id: string, // Interface Id to update
+  request: TransactionRequest, // Props must include `id` and `interfaceId`
+  errors?: { errors: ExecuteTxnUIErrors }, // Optional partial props for error handling or overrides
+) {
+  const {
+    signer,
+    chainId,
+    networkName,
+    maxFee,
+    calls,
+    selectedFeeToken,
+    includeDeploy,
+  } = request;
+  // Perform the interface update
+  await snap.request({
+    method: 'snap_updateInterface',
+    params: {
+      id,
+      ui: (
+        <ExecuteTxnUI
+          signer={signer}
+          chainId={chainId}
+          networkName={networkName}
+          maxFee={maxFee}
+          calls={calls}
+          selectedFeeToken={selectedFeeToken}
+          includeDeploy={includeDeploy}
+          {...errors}
+        />
+      ),
+    },
+  });
+}
+
+export type UpdateInterfaceParams = {
+  id: string;
+  ui: JSX.Element;
+};
+
+/**
+ *
+ * @param id
+ * @param ui
+ */
+export async function updateInterface(
+  id: string,
+  ui: JSX.Element,
+): Promise<void> {
+  await snap.request({
+    method: 'snap_updateInterface',
+    params: {
+      id,
+      ui,
+    },
+  });
+}
+
+/**
+ * Checks if the provided address has sufficient funds in either STRK or ETH to cover
+ * the total amount required for a set of contract calls and a suggested transaction fee.
+ *
+ * @param address - The wallet address to check balances for.
+ * @param network - The network object, which includes information such as chainId.
+ * @param calls - An array of contract call objects.
+ * Each object should contain a contract address and an optional amount.
+ * @param feeToken - Specifies the fee token type (either STRK or ETH) to use for the transaction fee.
+ * @param suggestedMaxFee - The maximum transaction fee to consider, as a stringified bigint.
+ * @returns Resolves to `true` if there are sufficient funds for the specified fee token
+ * to cover both the total amounts in `calls` and the `suggestedMaxFee`; otherwise, `false`.
+ */
+export async function hasSufficientFunds(
```

Comment:
look like we can simplify it, instead of doing all if else then for strk/eth

as we only need to consider if the user exceed their balance for 1 fee token
 (if you wanna to check the user has sufficient balance for what he is going to send, is another story, as he can send dai, abc, def)

e.g
```typescript
aysnc function hasSufficientFundsForFee({
  feeToken,
  network,
  address,
  calls,
}: {
  feeToken: Token, 
  network: Network, 
  address:string
  calls: FormattedCallData[]
}) {
    const balanceForFeeToken = await getBalance(
        address,
        feeToken.address,
        network,
        BlockIdentifierEnum.Pending,
      )
    : '0';
    
    // Calculate total STRK or ETH amounts from `calls`
    const totalSpendForFeeToken = calls.reduce((acc, call) => {
      const { tokenTransferData , contractAddress } = call
      if (tokenTransferData && contractAddress === feeToken.address) {
        acc += BigInt(tokenTransferData.amount);
      }
      return acc
    }, BigInt(suggestedMaxFee)) 

    return totalSpendForFeeToken >= balanceForFeeToken;
}
```